### ISSUE TITLE
Window maximize state not remembered

### ISSUE THREAD

**Issue by samuelznewton** [ransome1/sleek#110]
**Describe the bug**
Per [issue 79](https://github.com/ransome1/sleek/issues/79) and the release notes for [Release 0.9.8](https://github.com/ransome1/sleek/releases/tag/v0.9.8), 0.9.8 made sleek remember whether a window was maximized upon restart.  0.9.8 did indeed add this feature.  However, 1.0.0 has somewhat broken it.  Here's a description of the differences between the versions.

- 0.9.7: window resizing would be remembered between restarts.  However, maximizing the window did not count as resizing, and a restart would return the window to whatever size it was before it was maximized.
- 0.9.8 - 0.9.9: window resizing would be remembered between restarts.  sleek also remembered whether the window was maximized.  These were remembered separately.  If, for example, you had sleek open at 1000x1000 pixels, then you maximized it, then you closed it; when you reopened it, it would be maximized.  If you un-maximized it, it would go back to 1000x1000.  **This behavior, from 0.9.8 and 0.9.9, is desirable.**
- 1.0.0 - 1.0.2: window size is remembered between restarts.  A maximized window is considered to be the size of the screen.  However, sleek does not remember whether the window was maximized.  A window that was maximized will be placed a bit below the top of the screen upon restart.

**To Reproduce**
1. Open sleek
2. maximize it
3. close it
4. reopen it
5. notice that it's not maximized and looks something like this:
![image](https://user-images.githubusercontent.com/57228129/116586120-bb173b00-a8de-11eb-8046-9f9ed00781bd.png)

**Do you see any error entries in sleeks developer tools?**
No

**Expected behavior**
If sleek is maximized upon close, it ought to be maximized upon restart.  Remembering the unmaximized size as well is also desirable.  0.9.8 and 0.9.9 have desirable behavior.

**Desktop (please complete the following information):**
 - OS: Pop!_OS 20.10
 - Version of sleek: 1.0.0 - 1.0.2
 - Source: My main installation is Flathub, but I tested various versions to write this report using AppImages.

**Comment by ransome1** [comment]
Thank you for the report @samuelznewton, I'll take a look into it and try to fix it with the next release.

**Comment by ransome1** [comment]
Hi @samuelznewton. This one sounds simple, but it turns out it brought some obstacles with it. I improved the code and gave it testing on my Linux machine and a Windows VM. Can I ask you to give this development release a try and let me know if the issue still persists for you?

https://github.com/ransome1/sleek/releases/tag/v1.0.3

**Comment by samuelznewton** [comment]
Yep!  Window maximize state is remembered in this release.  Interestingly, this seems to have had an effect on the behavior of earlier versions as well:

- 0.9.9 and earlier will now always start in fullscreen.
- 1.0.0 - 1.0.2 will now start in the latest maximized state of 1.0.3

I don't think this is anything that needs fixing, I just found it strange.  There is a minor related issue that I've just described in [issue 113](https://github.com/ransome1/sleek/issues/113), but the issue of remembering maximize state does seem to be fixed.

Thanks for your quick work as usual! :)
